File: X42.java
public class X42 {
  public static void main(String[] args) throws Exception {
    if (args.length > 0 && args[0].equals("42")) {
      System.err.println("It depends!");
      System.exit(42);
    }
    System.out.println("What is the meaning of life?");
  }
}

File: X42.cpp
#include <iostream>
using namespace std;

int main(int argc, char *argv[]) {
  if (argc > 1 && std::string(argv[1]) == "42") {
    std::cerr << "It depends!\n";
    return 42;
  }
  std::cout << "What is the meaning of life?\n";
  return 0;
}
